menelaus/data_drift/psi_detector.py

```diff
@@ -0,0 +1,128 @@
+from menelaus.detector import BatchDetector
+import pandas as pd
+import numpy as np
+import copy
+
+
+class PSI_Detector(BatchDetector):
+    input_type = "batch"
+
+    def __init__(
+        self,
+    ):
+        # This initializes batch detector's parent class
+        super().__init__()
+
+        # Initialize any parameters to be used in this algorithm
+
+    def set_reference(self, X, y_true=None, y_pred=None):
+        # leave this, it uses the parent class to validate input
+        # and sets the self.reference variable to refer to the reference dataset
+
+        X, _, _ = super()._validate_input(X, None, None)
+        X = pd.DataFrame(X, columns=self._input_cols)
```

Comment:
- [ ] Ideally, replace the internal dataframe with a numpy array, but could be put off as an optimization.